simplekv/net/azurestore.py

```diff
@@ -128,3 +124,55 @@ def pickle_azure_store(store):
                              store.create_if_missing)
 
 copyreg.pickle(AzureBlockBlobStore, pickle_azure_store)
+
+
+class IOInterface(io.BufferedIOBase):
+    """
+    Class which provides a file-like interface to selectively read from a blob in the blob store.
+    """
+    def __init__(self, block_blob_service, container_name, key):
+        super(IOInterface, self).__init__()
+        self.block_blob_service = block_blob_service
+        self.container_name = container_name
+        self.key = key
+
+        blob = self.block_blob_service.get_blob_properties(container_name, key)
+        self.size = blob.properties.content_length
+        self.pos = 0
+
+    def tell(self):
+        """Returns he current offset as int. Always >= 0."""
+        return self.pos
+
+    def read(self, size=-1):
+        """Returns 'size' amount of bytes or less if there is no more data.
+        If no size is given all data is returned. size can be >= 0."""
+        with map_azure_exceptions(key=self.key):
+            if size < 0:
+                size = self.size - self.pos
+
+            end = min(self.pos + size - 1, self.size)
+            b = self.block_blob_service.get_blob_to_bytes(self.container_name,
+                                                          self.key,
```

Comment:
Formatting nitpick based on https://github.com/twitter/commons/blob/master/src/python/twitter/common/styleguide.md#best-practices 

It is commonly a good idea to stick to a common indentation rather than indenting up to the opening paranthesis as it offers more consistent style in larger files when function with different length are called 
```
 b = self.block_blob_service.get_blob_to_bytes(
        self.container_name,
        self.key,
        start_range=self.pos,
        end_range=end)  # end_range is inclusive
```

This is an external repo so feel free to ignore if you feel otherwise :)